docs/topics/media-pipeline.rst

```diff
@@ -258,13 +258,29 @@ For self-hosting you also might feel the need not to use SSL and not to verify S
 
     AWS_USE_SSL = False # or True (None by default)
     AWS_VERIFY = False # or True (None by default)
-
+    
 .. _botocore: https://github.com/boto/botocore
 .. _canned ACLs: https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl
 .. _Minio: https://github.com/minio/minio
 .. _s3.scality: https://s3.scality.com/
 
 
+Custom Amazon S3 headers
+'''''''''''''''''''''''''
+
+To send custom headers to AWS S3 when using :class:`~scrapy.pipelines.files.FilesPipeline`
+or :class:`~scrapy.pipelines.images.ImagesPipeline`, follow these steps:
+
+#. Subclass :class:`~scrapy.pipelines.files.S3FilesStore`, 
+
+#. Extend its ``HEADERS`` class attribute in your subclass to define the headers you want with the values you want. For reference,  `S3 <https://botocore.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html>`_ , `Common Request Headers <https://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonRequestHeaders.html>`_ , `Common Response Headers <https://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonResponseHeaders.html>`_ , `Request Response Data Mapping <https://docs.aws.amazon.com/apigateway/latest/developerguide/request-response-data-mappings.html>`_ . 
+You can see the header-to-key mapping in the source code of the class for additional header names.
```

Comment:
The second paragraph, because there is no empty line in between, will get rendered as an additional line of the first paragraph. i.e. we need an empty line.

Are all those links relevant here? I would imagine response headers are not or relevance.

When indicating that keader-to-key mapping can be seen in the source code, we should probably include a link to that source code, to make the user experience easier.

:lipstick:  It would be great if we split this into multiple, shorter lines to make it more readable when reading the source code.